tests: subsys/logging/log_syst : failed to build on rpi_pico

**Describe the bug**
Failed to build tests/subsys/logging/log_syst/ on rpi_pico

**To Reproduce**
Steps to reproduce the behavior:
1. twister -p rpi_pico -T tests/subsys/logging/log_syst/
2. see error

**Logs and console output**
```
Loading Zephyr default modules (Zephyr base).
-- Application: /home/guo/zephyrproject/zephyr/tests/subsys/logging/log_syst
-- Found Python3: /usr/bin/python3.8 (found suitable exact version "3.8.10") found components: Interpreter 
-- Cache files will be written to: /home/guo/.cache/zephyr
-- Zephyr version: 3.1.0-rc2 (/home/guo/zephyrproject/zephyr)
-- Found west (found suitable version "0.12.0", minimum required is "0.7.1")
-- Board: rpi_pico
-- ZEPHYR_TOOLCHAIN_VARIANT not set, trying to locate Zephyr SDK
-- Found host-tools: zephyr 0.14.2 (/home/guo/zephyr-sdk-0.14.2)
-- Found dtc: /home/guo/zephyr-sdk-0.14.2/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.6.0", minimum required is "1.4.6")
-- Found toolchain: zephyr 0.14.2 (/home/guo/zephyr-sdk-0.14.2)
-- Found BOARD.dts: /home/guo/zephyrproject/zephyr/boards/arm/rpi_pico/rpi_pico.dts
-- Generated zephyr.dts: /home/guo/zephyrproject/zephyr/twister-out/rpi_pico/tests/subsys/logging/log_syst/logging.log_syst.mipi_syst/zephyr/zephyr.dts
-- Generated devicetree_unfixed.h: /home/guo/zephyrproject/zephyr/twister-out/rpi_pico/tests/subsys/logging/log_syst/logging.log_syst.mipi_syst/zephyr/include/generated/devicetree_unfixed.h
-- Generated device_extern.h: /home/guo/zephyrproject/zephyr/twister-out/rpi_pico/tests/subsys/logging/log_syst/logging.log_syst.mipi_syst/zephyr/include/generated/device_extern.h
-- Including generated dts.cmake file: /home/guo/zephyrproject/zephyr/twister-out/rpi_pico/tests/subsys/logging/log_syst/logging.log_syst.mipi_syst/zephyr/dts.cmake
/home/guo/zephyrproject/zephyr/twister-out/rpi_pico/tests/subsys/logging/log_syst/logging.log_syst.mipi_syst/zephyr/zephyr.dts:60.34-65.5: Warning (simple_bus_reg): /soc/peripheral-clk: missing or empty reg/ranges property
/home/guo/zephyrproject/zephyr/twister-out/rpi_pico/tests/subsys/logging/log_syst/logging.log_syst.mipi_syst/zephyr/zephyr.dts:66.26-71.5: Warning (simple_bus_reg): /soc/system-clk: missing or empty reg/ranges property
/home/guo/zephyrproject/zephyr/twister-out/rpi_pico/tests/subsys/logging/log_syst/logging.log_syst.mipi_syst/zephyr/zephyr.dts:81.36-112.5: Warning (unique_unit_address_if_enabled): /soc/pin-controller@40014000: duplicate unit-address (also used in node /soc/gpio@40014000)
Parsing /home/guo/zephyrproject/zephyr/tests/subsys/logging/log_syst/Kconfig
Loaded configuration '/home/guo/zephyrproject/zephyr/boards/arm/rpi_pico/rpi_pico_defconfig'
Merged configuration '/home/guo/zephyrproject/zephyr/tests/subsys/logging/log_syst/prj.conf'
Merged configuration '/home/guo/zephyrproject/zephyr/twister-out/rpi_pico/tests/subsys/logging/log_syst/logging.log_syst.mipi_syst/twister/testsuite_extra.conf'
Configuration saved to '/home/guo/zephyrproject/zephyr/twister-out/rpi_pico/tests/subsys/logging/log_syst/logging.log_syst.mipi_syst/zephyr/.config'
Kconfig header saved to '/home/guo/zephyrproject/zephyr/twister-out/rpi_pico/tests/subsys/logging/log_syst/logging.log_syst.mipi_syst/zephyr/include/generated/autoconf.h'
-- The C compiler identification is GNU 10.3.0
-- The CXX compiler identification is GNU 10.3.0
-- The ASM compiler identification is GNU
-- Found assembler: /home/guo/zephyr-sdk-0.14.2/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc
-- Configuring done
-- Generating done
-- Build files have been written to: /home/guo/zephyrproject/zephyr/twister-out/rpi_pico/tests/subsys/logging/log_syst/logging.log_syst.mipi_syst
[  1%] Generating misc/generated/syscalls.json, misc/generated/struct_tags.json
[  2%] Built target parse_syscalls_target
[  2%] Generating include/generated/kobj-types-enum.h, include/generated/otype-to-str.h, include/generated/otype-to-size.h
[  2%] Built target kobj_types_h_target
[  2%] Generating include/generated/syscall_dispatch.c, include/generated/syscall_list.h
[  2%] Built target syscall_list_h_target
[  3%] Generating include/generated/driver-validation.h
[  3%] Built target driver_validation_h_target
[  3%] Building C object zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj
[  3%] Built target offsets
[  4%] Generating include/generated/offsets.h
[  4%] Built target offsets_h
[  5%] Generating include/generated/version.h
-- Zephyr version: 3.1.0-rc2 (/home/guo/zephyrproject/zephyr), build: v3.1.0-rc2-18-gea41f4ec8795
[  5%] Built target version_h
[  5%] Built target zephyr_generated_headers
[  5%] Building C object CMakeFiles/app.dir/src/main.c.obj
[  6%] Building C object CMakeFiles/app.dir/src/mock_backend.c.obj
[  6%] Linking C static library app/libapp.a
[  6%] Built target app
[  6%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/cbprintf.c.obj
[  7%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/cbprintf_packaged.c.obj
[  8%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/crc32c_sw.c.obj
[  8%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/crc32_sw.c.obj
[  9%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/crc16_sw.c.obj
[  9%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/crc8_sw.c.obj
[ 10%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/crc7_sw.c.obj
[ 10%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/dec.c.obj
[ 11%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/fdtable.c.obj
[ 12%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/hex.c.obj
[ 12%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/notify.c.obj
[ 13%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/printk.c.obj
[ 13%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/onoff.c.obj
[ 14%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/rb.c.obj
[ 15%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/sem.c.obj
[ 15%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/thread_entry.c.obj
[ 16%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/timeutil.c.obj
[ 16%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/heap.c.obj
[ 17%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/heap-validate.c.obj
[ 18%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/bitarray.c.obj
[ 18%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/multi_heap.c.obj
[ 19%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/cbprintf_complete.c.obj
[ 19%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/assert.c.obj
[ 20%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/mpsc_pbuf.c.obj
[ 21%] Building C object zephyr/CMakeFiles/zephyr.dir/misc/generated/configs.c.obj
[ 21%] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/logging/log_list.c.obj
[ 22%] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/logging/log_core.c.obj
[ 22%] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/logging/log_mgmt.c.obj
[ 23%] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/logging/log_msg.c.obj
[ 25%] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/logging/log_output.c.obj
[ 25%] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/logging/log_msg2.c.obj
[ 26%] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/logging/log_output_syst.c.obj
[ 26%] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/tracing/tracing_none.c.obj
[ 27%] Linking C static library libzephyr.a
[ 27%] Built target zephyr
[ 27%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/main_weak.c.obj
[ 28%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/banner.c.obj
[ 29%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/device.c.obj
[ 29%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/errno.c.obj
[ 30%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/fatal.c.obj
[ 30%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/init.c.obj
[ 31%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/kheap.c.obj
[ 32%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/mem_slab.c.obj
[ 32%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/thread.c.obj
[ 33%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/version.c.obj
[ 33%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/idle.c.obj
[ 34%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/mailbox.c.obj
[ 35%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/msg_q.c.obj
[ 35%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/mutex.c.obj
[ 36%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/pipes.c.obj
[ 36%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/queue.c.obj
[ 37%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/sem.c.obj
[ 38%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/stack.c.obj
[ 38%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/system_work_q.c.obj
[ 39%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/work.c.obj
[ 39%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/sched.c.obj
[ 40%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/condvar.c.obj
[ 41%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/xip.c.obj
[ 41%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/timeout.c.obj
[ 42%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/timer.c.obj
[ 42%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/atomic_c.c.obj
[ 43%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/mempool.c.obj
[ 44%] Linking C static library libkernel.a
[ 44%] Built target kernel
[ 44%] Generating linker_zephyr_pre0.cmd
[ 44%] Built target linker_zephyr_pre0_script
[ 44%] Building C object zephyr/arch/common/CMakeFiles/isr_tables.dir/isr_tables.c.obj
[ 45%] Linking C static library libisr_tables.a
[ 45%] Built target isr_tables
[ 46%] Building C object zephyr/arch/common/CMakeFiles/arch__common.dir/sw_isr_common.c.obj
[ 46%] Linking C static library libarch__common.a
[ 46%] Built target arch__common
Scanning dependencies of target arch__arm__core__aarch32
[ 47%] Building ASM object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/cpu_idle.S.obj
[ 48%] Building C object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/fatal.c.obj
[ 48%] Building C object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/irq_manage.c.obj
[ 50%] Building C object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/nmi.c.obj
[ 50%] Building ASM object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/nmi_on_reset.S.obj
[ 51%] Building C object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/prep_c.c.obj
[ 52%] Building C object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/swap.c.obj
[ 52%] Building ASM object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/swap_helper.S.obj
[ 53%] Building C object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/thread.c.obj
[ 53%] Building ASM object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/isr_wrapper.S.obj
[ 54%] Linking C static library libarch__arm__core__aarch32.a
[ 54%] Built target arch__arm__core__aarch32
Scanning dependencies of target arch__arm__core__aarch32__cortex_m
[ 55%] Building ASM object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/exc_exit.S.obj
[ 55%] Building C object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/fault.c.obj
[ 56%] Building ASM object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/fault_s.S.obj
[ 56%] Building C object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/fpu.c.obj
[ 57%] Building C object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/irq_init.c.obj
[ 58%] Building ASM object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/reset.S.obj
[ 58%] Building C object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/scb.c.obj
[ 59%] Building C object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/thread_abort.c.obj
[ 59%] Building ASM object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/vector_table.S.obj
[ 60%] Linking C static library libarch__arm__core__aarch32__cortex_m.a
[ 60%] Built target arch__arm__core__aarch32__cortex_m
[ 61%] Building C object zephyr/lib/libc/newlib/CMakeFiles/lib__libc__newlib.dir/libc-hooks.c.obj
[ 61%] Linking C static library liblib__libc__newlib.a
[ 61%] Built target lib__libc__newlib
[ 62%] Building C object zephyr/lib/posix/CMakeFiles/lib__posix.dir/pthread_common.c.obj
[ 62%] Building C object zephyr/lib/posix/CMakeFiles/lib__posix.dir/nanosleep.c.obj
[ 63%] Linking C static library liblib__posix.a
[ 63%] Built target lib__posix
[ 63%] Building C object zephyr/soc/arm/rpi_pico/rp2/CMakeFiles/soc__arm__rpi_pico__rp2.dir/soc.c.obj
[ 64%] Linking C static library libsoc__arm__rpi_pico__rp2.a
[ 64%] Built target soc__arm__rpi_pico__rp2
[ 64%] Building C object zephyr/subsys/testsuite/ztest/CMakeFiles/subsys__testsuite__ztest.dir/src/ztest.c.obj
[ 65%] Building C object zephyr/subsys/testsuite/ztest/CMakeFiles/subsys__testsuite__ztest.dir/src/ztest_error_hook.c.obj
[ 66%] Linking C static library libsubsys__testsuite__ztest.a
[ 66%] Built target subsys__testsuite__ztest
[ 66%] Building C object zephyr/drivers/console/CMakeFiles/drivers__console.dir/uart_console.c.obj
[ 67%] Linking C static library libdrivers__console.a
[ 67%] Built target drivers__console
[ 67%] Building C object zephyr/drivers/gpio/CMakeFiles/drivers__gpio.dir/gpio_rpi_pico.c.obj
[ 68%] Linking C static library libdrivers__gpio.a
[ 68%] Built target drivers__gpio
[ 69%] Building C object zephyr/drivers/serial/CMakeFiles/drivers__serial.dir/uart_rpi_pico.c.obj
In file included from /home/guo/zephyrproject/zephyr/drivers/serial/uart_rpi_pico.c:9:
/home/guo/zephyrproject/zephyr/include/zephyr/drivers/reset.h: In function 'z_impl_reset_assert':
/home/guo/zephyrproject/zephyr/include/zephyr/drivers/reset.h:213:28: error: macro "assert" passed 2 arguments, but takes just 1
  213 |  return api->assert(dev, id);
      |                            ^
In file included from /home/guo/zephyr-sdk-0.14.2/arm-zephyr-eabi/arm-zephyr-eabi/sys-include/sys/reent.h:503,
                 from /home/guo/zephyr-sdk-0.14.2/arm-zephyr-eabi/arm-zephyr-eabi/sys-include/sys/errno.h:11,
                 from /home/guo/zephyr-sdk-0.14.2/arm-zephyr-eabi/arm-zephyr-eabi/sys-include/errno.h:9,
                 from /home/guo/zephyrproject/zephyr/include/zephyr/drivers/uart.h:23,
                 from /home/guo/zephyrproject/zephyr/drivers/serial/uart_rpi_pico.c:8:
/home/guo/zephyr-sdk-0.14.2/arm-zephyr-eabi/arm-zephyr-eabi/sys-include/assert.h:16: note: macro "assert" defined here
   16 | # define assert(__e) ((__e) ? (void)0 : __assert_func (__FILE__, __LINE__, \
      | 
In file included from /home/guo/zephyrproject/zephyr/drivers/serial/uart_rpi_pico.c:9:
/home/guo/zephyrproject/zephyr/include/zephyr/drivers/reset.h:213:12: error: returning 'reset_api_assert' {aka 'int (* const)(const struct device *, unsigned int)'} from a function with return type 'int' makes integer from pointer without a cast [-Werror=int-conversion]
  213 |  return api->assert(dev, id);
      |         ~~~^~~~~~~~
cc1: all warnings being treated as errors
make[2]: *** [zephyr/drivers/serial/CMakeFiles/drivers__serial.dir/build.make:76: zephyr/drivers/serial/CMakeFiles/drivers__serial.dir/uart_rpi_pico.c.obj] Error 1
make[1]: *** [CMakeFiles/Makefile2:3276: zephyr/drivers/serial/CMakeFiles/drivers__serial.dir/all] Error 2
make: *** [Makefile:91: all] Error 2
```

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain 0.14.2

--- a/doc/hardware/peripherals/reset.rst
+++ b/doc/hardware/peripherals/reset.rst
@@ -12,8 +12,8 @@ control over their reset input signals, including the ability to assert,
 deassert and toggle those signals. Also, the reset status of the reset input
 signal can be checked.
 
-Mainly, the assert and deassert API functions are optional because in most
-cases we want to toggle the reset signals.
+Mainly, the line_assert and line_deassert API functions are optional because
+in most cases we want to toggle the reset signals.
 
 Configuration Options
 *********************

--- a/doc/releases/release-notes-3.1.rst
+++ b/doc/releases/release-notes-3.1.rst
@@ -294,6 +294,10 @@ Drivers and Sensors
     easier to use the PWM API when the PWM channel, period and flags are taken
     from a Devicetree PWM cell.
 
+* Reset
+
+  * Added reset controller driver API.
+
 * Sensor
 
 * Serial

--- a/drivers/reset/reset_rpi_pico.c
+++ b/drivers/reset/reset_rpi_pico.c
@@ -106,26 +106,26 @@ static int reset_rpi_update(const struct device *dev, uint32_t id, uint8_t asser
 	return reset_rpi_write_register(dev, offset, value);
 }
 
-static int reset_rpi_assert(const struct device *dev, uint32_t id)
+static int reset_rpi_line_assert(const struct device *dev, uint32_t id)
 {
 	return reset_rpi_update(dev, id, 1);
 }
 
-static int reset_rpi_deassert(const struct device *dev, uint32_t id)
+static int reset_rpi_line_deassert(const struct device *dev, uint32_t id)
 {
 	return reset_rpi_update(dev, id, 0);
 }
 
-static int reset_rpi_toggle(const struct device *dev, uint32_t id)
+static int reset_rpi_line_toggle(const struct device *dev, uint32_t id)
 {
 	int ret;
 
-	ret = reset_rpi_assert(dev, id);
+	ret = reset_rpi_line_assert(dev, id);
 	if (ret) {
 		return ret;
 	}
 
-	return reset_rpi_deassert(dev, id);
+	return reset_rpi_line_deassert(dev, id);
 }
 
 static int reset_rpi_init(const struct device *dev)
@@ -137,9 +137,9 @@ static int reset_rpi_init(const struct device *dev)
 
 static const struct reset_driver_api reset_rpi_driver_api = {
 	.status = reset_rpi_status,
-	.assert = reset_rpi_assert,
-	.deassert = reset_rpi_deassert,
-	.toggle = reset_rpi_toggle,
+	.line_assert = reset_rpi_line_assert,
+	.line_deassert = reset_rpi_line_deassert,
+	.line_toggle = reset_rpi_line_toggle,
 };
 
 #define RPI_RESET_INIT(idx)							\

--- a/include/zephyr/drivers/reset.h
+++ b/include/zephyr/drivers/reset.h
@@ -114,32 +114,32 @@ typedef int (*reset_api_status)(const struct device *dev, uint32_t id, uint8_t *
 /**
  * API template to put the device in reset state.
  *
- * @see reset_assert
+ * @see reset_line_assert
  */
-typedef int (*reset_api_assert)(const struct device *dev, uint32_t id);
+typedef int (*reset_api_line_assert)(const struct device *dev, uint32_t id);
 
 /**
  * API template to take out the device from reset state.
  *
- * @see reset_deassert
+ * @see reset_line_deassert
  */
-typedef int (*reset_api_deassert)(const struct device *dev, uint32_t id);
+typedef int (*reset_api_line_deassert)(const struct device *dev, uint32_t id);
 
 /**
  * API template to reset the device.
  *
- * @see reset_toggle
+ * @see reset_line_toggle
  */
-typedef int (*reset_api_toggle)(const struct device *dev, uint32_t id);
+typedef int (*reset_api_line_toggle)(const struct device *dev, uint32_t id);
 
 /**
  * @brief Reset Controller driver API
  */
 __subsystem struct reset_driver_api {
 	reset_api_status status;
-	reset_api_assert assert;
-	reset_api_deassert deassert;
-	reset_api_toggle toggle;
+	reset_api_line_assert line_assert;
+	reset_api_line_deassert line_deassert;
+	reset_api_line_toggle line_toggle;
 };
 
 /** @endcond */
@@ -200,17 +200,17 @@ static inline int reset_status_dt(const struct reset_dt_spec *spec, uint8_t *sta
  * @retval -ENOSYS If the functionality is not implemented by the driver.
  * @retval -errno Other negative errno in case of failure.
  */
-__syscall int reset_assert(const struct device *dev, uint32_t id);
+__syscall int reset_line_assert(const struct device *dev, uint32_t id);
 
-static inline int z_impl_reset_assert(const struct device *dev, uint32_t id)
+static inline int z_impl_reset_line_assert(const struct device *dev, uint32_t id)
 {
 	const struct reset_driver_api *api = (const struct reset_driver_api *)dev->api;
 
-	if (api->assert == NULL) {
+	if (api->line_assert == NULL) {
 		return -ENOSYS;
 	}
 
-	return api->assert(dev, id);
+	return api->line_assert(dev, id);
 }
 
 /**
@@ -218,15 +218,15 @@ static inline int z_impl_reset_assert(const struct device *dev, uint32_t id)
  *
  * This is equivalent to:
  *
- *     reset_assert(spec->dev, spec->id);
+ *     reset_line_assert(spec->dev, spec->id);
  *
  * @param spec Reset controller specification from devicetree
  *
- * @return a value from reset_assert()
+ * @return a value from reset_line_assert()
  */
-static inline int reset_assert_dt(const struct reset_dt_spec *spec)
+static inline int reset_line_assert_dt(const struct reset_dt_spec *spec)
 {
-	return reset_assert(spec->dev, spec->id);
+	return reset_line_assert(spec->dev, spec->id);
 }
 
 /**
@@ -242,17 +242,17 @@ static inline int reset_assert_dt(const struct reset_dt_spec *spec)
  * @retval -ENOSYS If the functionality is not implemented by the driver.
  * @retval -errno Other negative errno in case of failure.
  */
-__syscall int reset_deassert(const struct device *dev, uint32_t id);
+__syscall int reset_line_deassert(const struct device *dev, uint32_t id);
 
-static inline int z_impl_reset_deassert(const struct device *dev, uint32_t id)
+static inline int z_impl_reset_line_deassert(const struct device *dev, uint32_t id)
 {
 	const struct reset_driver_api *api = (const struct reset_driver_api *)dev->api;
 
-	if (api->deassert == NULL) {
+	if (api->line_deassert == NULL) {
 		return -ENOSYS;
 	}
 
-	return api->deassert(dev, id);
+	return api->line_deassert(dev, id);
 }
 
 /**
@@ -260,15 +260,15 @@ static inline int z_impl_reset_deassert(const struct device *dev, uint32_t id)
  *
  * This is equivalent to:
  *
- *     reset_deassert(spec->dev, spec->id)
+ *     reset_line_deassert(spec->dev, spec->id)
  *
  * @param spec Reset controller specification from devicetree
  *
- * @return a value from reset_deassert()
+ * @return a value from reset_line_deassert()
  */
-static inline int reset_deassert_dt(const struct reset_dt_spec *spec)
+static inline int reset_line_deassert_dt(const struct reset_dt_spec *spec)
 {
-	return reset_deassert(spec->dev, spec->id);
+	return reset_line_deassert(spec->dev, spec->id);
 }
 
 /**
@@ -283,17 +283,17 @@ static inline int reset_deassert_dt(const struct reset_dt_spec *spec)
  * @retval -ENOSYS If the functionality is not implemented by the driver.
  * @retval -errno Other negative errno in case of failure.
  */
-__syscall int reset_toggle(const struct device *dev, uint32_t id);
+__syscall int reset_line_toggle(const struct device *dev, uint32_t id);
 
-static inline int z_impl_reset_toggle(const struct device *dev, uint32_t id)
+static inline int z_impl_reset_line_toggle(const struct device *dev, uint32_t id)
 {
 	const struct reset_driver_api *api = (const struct reset_driver_api *)dev->api;
 
-	if (api->toggle == NULL) {
+	if (api->line_toggle == NULL) {
 		return -ENOSYS;
 	}
 
-	return api->toggle(dev, id);
+	return api->line_toggle(dev, id);
 }
 
 /**
@@ -301,15 +301,15 @@ static inline int z_impl_reset_toggle(const struct device *dev, uint32_t id)
  *
  * This is equivalent to:
  *
- *     reset_toggle(spec->dev, spec->id)
+ *     reset_line_toggle(spec->dev, spec->id)
  *
  * @param spec Reset controller specification from devicetree
  *
- * @return a value from reset_toggle()
+ * @return a value from reset_line_toggle()
  */
-static inline int reset_toggle_dt(const struct reset_dt_spec *spec)
+static inline int reset_line_toggle_dt(const struct reset_dt_spec *spec)
 {
-	return reset_toggle(spec->dev, spec->id);
+	return reset_line_toggle(spec->dev, spec->id);
 }
 
 /**

